packages/backend/scripts/diffc.py

```diff
@@ -0,0 +1,241 @@
+    subpath2, logical_name2 = get_project_subpath(
+        base_path, folder2, dummy_directory)
+
+    # Iterate over directories
+    for directory in common_directories:
+        no_changes = True
+        directory_title = " {0} ".format(directory)
+        num_dashes = (terminal_width - len(directory_title)) // 2
+        print(f"\n{'-' * num_dashes}{directory_title}{'-' * num_dashes}\n")
+
+        # Construct the directory paths for the current directory
+        dir_path1 = os.path.join(
+            base_path, "discovery", folder1, "ethereum", ".code", directory, subpath1)
+        dir_path2 = os.path.join(
+            base_path, "discovery", folder2, "ethereum", ".code", directory, subpath2)
+
+        if not os.path.exists(dir_path1):
+            print(
+                f"{folder1} does not contain {directory} with project structure {logical_name1}.")
+            continue
+        if not os.path.exists(dir_path2):
+            print(
+                f"{folder2} does not contain {directory} with project structure {logical_name2}.")
+            continue
+
+        sol_files1 = [os.path.join(root, f) for root, dirs, files in os.walk(
+            dir_path1) for f in files if f.endswith('.sol')]
+        sol_files2 = [os.path.join(root, f) for root, dirs, files in os.walk(
+            dir_path2) for f in files if f.endswith('.sol')]
+
+        # Map to common format
+        sol_files1 = {f.replace(dir_path1, "<placeholder>")
+                      for f in sol_files1}
+        sol_files2 = {f.replace(dir_path2, "<placeholder>")
+                      for f in sol_files2}
+
+        common_sol_files = sol_files1 & sol_files2
+
+        for sol_file in common_sol_files:
+            file1 = sol_file.replace(
+                "<placeholder>", dir_path1).replace(folder1, folder1)
+            file2 = sol_file.replace(
+                "<placeholder>", dir_path2).replace(folder1, folder2)
+
+            # Run "forge fmt" on each file before diffing
+            # subprocess.run(["forge", "fmt", file1])
+            # subprocess.run(["forge", "fmt", file2])
+
+            pid, fd = pty.fork()
+            if pid == 0:  # child process
+                os.execvp("difft", ["difft", "--skip-unchanged",
+                          "--ignore-comments", "--width", str(terminal_width), file1, file2])
+            else:  # parent process
+                result = b""
+                while True:
+                    try:
+                        chunk = os.read(fd, 1024)
+                        if not chunk:  # no more data
+                            break
+                        result += chunk
+                    except OSError:
+                        break
+                result = result.decode()
+                if "No changes." not in result:
+                    # Split result into lines
+                    lines = result.split('\n')
+                    # Only print lines that are not empty
```

Comment:
This is my last nitpick about comments, I promise. This comment is redundant with the second one just below. You can understand what is happening by reading either of these comments, so two comments are twice as likely to become outdated when this code changes. Or worse, one will get updated and the second won't and now you have a contradiction in your comments and the reader does not know which one to believe in